addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,138 @@
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<bridge-type id="account">
+		<label>iCloud Account</label>
+		<description>Provides various information for iCloud devices registered with this account.</description>
+
+		<channels>
+			<channel id="owner" typeId="Owner" />
+			<channel id="numberOfDevices" typeId="NumberOfDevices" />
+			<channel id="refresh" typeId="Refresh" />
+		</channels>
+
+		<config-description>
+			<parameter name="appleId" type="text" required="true">
+				<context>email</context>
+				<label>Apple Id</label>
+				<description>Apple Id (e-mail) to access iCloud information.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Apple iCloud password for the given Id.</description>
+			</parameter>
+			<parameter name="refreshTimeInMinutes" type="integer" min="5" max="65535" unit="min">
+				<label>Refresh</label>
+				<description>Refresh time in minutes.</description>
+				<default>5</default>
+			</parameter>
+			<parameter name="googleAPIKey" type="text">
+				<label>Google API Key</label>
+				<description>Google API key to be used for address lookup calls (to translate coordinates into a human readable string). Without a key the number of daily request are limited and the bindings address lookup functionality will stop working.</description>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>iCloud Device</label>
+
+		<channels>
+			<channel typeId="BatteryStatus" id="batteryStatus" />
+			<channel typeId="system.battery-level" id="batteryLevel" />
+			<channel typeId="FindMyPhone" id="findMyPhone" />
+			<channel typeId="Location" id="location" />
+			<channel typeId="LocationAccuracy" id="locationAccuracy" />
+			<channel typeId="LocationLastUpdate" id="locationLastUpdate" />
+			<channel typeId="DistanceFromHome" id="distanceFromHome" />
+			<channel typeId="AddressStreet" id="addressStreet" />
+			<channel typeId="AddressCity" id="addressCity" />
+			<channel typeId="AddressCountry" id="addressCountry" />
+			<channel typeId="FormattedAddress" id="formattedAddress" />
+			<channel typeId="DeviceName" id="deviceName" />
+		</channels>
+
+		<representation-property>deviceId</representation-property>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device id</label>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="NumberOfDevices">
+		<item-type>Number</item-type>
+		<label>Number of devices</label>
+		<description>Number of registered devices with this iCloud account.</description>
+	</channel-type>
+	<channel-type id="Owner">

```

Comment:
This should not be a channel as it won't change. You can add it as a property to the Bridge, if you like.

Reply:
Can a property also be  accessed by rules/sitemaps; as for installations with multiple accounts this might be of use.